src/sentry/buffer/redis.py

```diff
@@ -144,8 +146,13 @@ def process_pending(self, partition=None):
         if partition is None and self.pending_partitions > 1:
             # If we're using partitions, this one task fans out into
             # N subtasks instead.
+            now = timezone.now()
             for i in range(self.pending_partitions):
-                process_pending.apply_async(kwargs={'partition': i})
+                process_pending.apply_async(
+                    kwargs={'partition': i},
+                    # Smear over 10 seconds
+                    eta=now + timedelta(seconds=(float(i) / (self.pending_partitions - 1)) * 10),
```

Comment:
Fun fact, `countdown` requires an integer in seconds, and you can't use fractions of a second. `eta` allows an explicit `datetime` object down to millisecond precision.

Reply:
docs aren't super clear about it either

> countdown (float) – Number of seconds into the future that the task should execute. Defaults to immediate execution.
> eta (datetime) – Absolute time and date of when the task should be executed. May not be specified if countdown is also supplied.